src/ucp/rndv/rndv_put.c

```diff
@@ -573,6 +573,21 @@ static void ucp_proto_rndv_put_mtype_frag_completion(uct_completion_t *uct_comp)
     ucp_proto_rndv_ppln_send_frag_complete(req, 1);
 }
 
+static ucs_memory_type_t
+ucp_proto_rndv_put_mtype_frag_mem_type(uint64_t rndv_frag_mem_types,
+                                       ucs_memory_type_t rkey_mem_type)
+{
+    if (rndv_frag_mem_types == 0) {
+        return UCS_MEMORY_TYPE_UNKNOWN;
```

Comment:
BTW, would the mtype protocol be disabled in this case? i'd expect it to be disabled completely. maybe better check it in the caller function ucp_proto_rndv_put_mtype_probe, and maybe even inline this function code inside ucp_proto_rndv_put_mtype_probe?

Reply:
Yes, it will be disabled, and `rndv/am` will be used as a fallback.